File: extensions/jetbrains/src/main/java/com/closecrab/plugin/CloseCrabClient.java
package com.closecrab.plugin;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class CloseCrabClient {
    private static final String DEFAULT_URL = "http://localhost:9001";
    private final HttpClient client = HttpClient.newHttpClient();
    private String serverUrl;

    public CloseCrabClient() {
        this.serverUrl = DEFAULT_URL;
    }

    public CloseCrabClient(String url) {
        this.serverUrl = url;
    }

    public String chat(String message) throws Exception {
        String json = "{\"message\":\"" + escapeJson(message) + "\",\"session_id\":\"jetbrains\"}";
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(serverUrl + "/chat"))
                .header("Content-Type", "application/json")
                .POST(HttpRequest.BodyPublishers.ofString(json))
                .build();
        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        return response.body();
    }

    public boolean isConnected() {
        try {
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(serverUrl + "/health"))
                    .GET().build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            return response.statusCode() == 200;
        } catch (Exception e) {
            return false;
        }
    }

    private String escapeJson(String s) {
        return s.replace("\\", "\\\\").replace("\"", "\\\"").replace("\n", "\\n");
    }
}

File: extensions/visualstudio/CloseCrab/CloseCrabClient.cs
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace CloseCrab
{
    public class CloseCrabClient
    {
        private static readonly HttpClient _client = new HttpClient();
        private readonly string _serverUrl;

        public CloseCrabClient(string serverUrl = "http://localhost:9001")
        {
            _serverUrl = serverUrl;
            _client.Timeout = TimeSpan.FromSeconds(120);
        }

        public async Task<string> ChatAsync(string message, string sessionId = "visualstudio")
        {
            var payload = new { message, session_id = sessionId };
            var json = JsonConvert.SerializeObject(payload);
            var content = new StringContent(json, Encoding.UTF8, "application/json");

            try
            {
                var response = await _client.PostAsync($"{_serverUrl}/chat", content);
                var body = await response.Content.ReadAsStringAsync();
                var result = JObject.Parse(body);
                return result["response"]?.ToString() ?? result["error"]?.ToString() ?? "No response";
            }
            catch (Exception ex)
            {
                return $"Error: {ex.Message}";
            }
        }

        public async Task<bool> IsConnectedAsync()
        {
            try
            {
                var response = await _client.GetAsync($"{_serverUrl}/health");
                return response.IsSuccessStatusCode;
            }
            catch { return false; }
        }
    }
}
